lib/apiservers/service/restapi/handlers/vch_cert_get.go

```diff
@@ -35,21 +35,23 @@ type VCHCertGet struct{}
 type VCHDatacenterCertGet struct{}
 
 func (h *VCHCertGet) Handle(params operations.GetTargetTargetVchVchIDCertificateParams, principal interface{}) middleware.Responder {
-	d, err := buildData(params.HTTPRequest.Context(),
-		url.URL{Host: params.Target},
-		principal.(Credentials).user,
-		principal.(Credentials).pass,
-		params.Thumbprint,
-		nil,
-		nil)
+	op := trace.NewOperation(params.HTTPRequest.Context(), "VCHCertGet: %s", params.VchID)
+
+	b := BuildDataParams{
+		url:        url.URL{Host: params.Target},
```

Comment:
Since you're adjusting this pattern anyway, maybe `BuildDataParams` should take a raw target so we can have the `url.URL{Host: params.Target}` happen in a single place (i.e., `buildData`).

Reply:
Good idea, fixed!